packages/notifications/src/inAppMessaging/providers/pinpoint/utils/processInAppMessages.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { InAppMessage, InAppMessagingEvent } from '../../../types';
+import {
+	InAppMessageCounts,
+	InAppMessageCountMap,
+	DailyInAppMessageCounter,
+} from '../types';
+import {
+	extractContent,
+	extractMetadata,
+	interpretLayout,
+	isBeforeEndDate,
+	matchesAttributes,
+	matchesEventType,
+	matchesMetrics,
+} from './helpers';
+import type { InAppMessageCampaign as PinpointInAppMessage } from '@aws-amplify/core/internals/aws-clients/pinpoint';
+import { ConsoleLogger } from '@aws-amplify/core/internals/utils';
+import { defaultStorage } from '@aws-amplify/core';
+
+const MESSAGE_DAILY_COUNT_KEY = 'pinpointProvider_inAppMessages_dailyCount';
+const MESSAGE_TOTAL_COUNT_KEY = 'pinpointProvider_inAppMessages_totalCount';
+const logger = new ConsoleLogger('InAppMessaging.processInAppMessages');
+
+const sessionMessageCountMap: InAppMessageCountMap = {};
+
+export async function processInAppMessages(
+	messages: PinpointInAppMessage[],
+	event: InAppMessagingEvent
+): Promise<InAppMessage[]> {
+	let highestPrioritySeen: number;
```

Comment:
Are we turning on ts strick mode for this package?

```suggestion
	let highestPrioritySeen: number | undefined;
```

Reply:
+1, we should just turn that on now before we migrate more APIs